components/property_info.rst

```diff
@@ -0,0 +1,429 @@
+provide :ref:`extensive data type information<components-property-info-type>` for a property.
+
+.. code-block:: php
+
+    $types = $propertyInfo->getTypes($class, $property);
+
+    /*
+      Example Result
+      --------------
+      array(1) {
+        [0] =>
+        class Symfony\Component\PropertyInfo\Type (6) {
+          private $builtinType          => string(6) "string"
+          private $nullable             => bool(false)
+          private $class                => NULL
+          private $collection           => bool(false)
+          private $collectionKeyType    => NULL
+          private $collectionValueType  => NULL
+        }
+      }
+    */
+
+Description Information
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Extractors that implement :class:`PropertyDescriptionExtractorInterface<Symfony\\Component\\PropertyInfo\\PropertyDescriptionExtractorInterface>`
+provide long and short descriptions from a properties annotations as strings.
+
+.. code-block:: php
+
+    $title = $propertyInfo->getShortDescription($class, $property);
+    /*
+      Example Result
+      --------------
+      string(41) "This is the first line of the DocComment."
+    */
+
+    $paragraph = $propertyInfo->getLongDescription($class, $property);
+    /*
+      Example Result
+      --------------
+      string(79):
+        These is the subsequent paragraph in the DocComment.
+        It can span multiple lines.
+    */
+
+Access Information
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Extractors that implement :class:`PropertyAccessExtractorInterface<Symfony\\Component\\PropertyInfo\\PropertyAccessExtractorInterface>`
+provide whether properties are readable or writable as booleans.
+
+.. code-block:: php
+
+    $propertyInfo->isReadable($class, $property);
+    // Example Result: bool(true)
+
+    $propertyInfo->isWritable($class, $property);
+    // Example Result: bool(false)
+
+.. tip::
+
+    The main :class:`PropertyInfoExtractor<Symfony\\Component\\PropertyInfo\\PropertyInfoExtractor>` class implements
+    all four interfaces, delgating the extraction of property information to the extractors that have been registered
+    with it.
+
+    This means that any method available on each of the extractors is also available on the main
+    :class:`PropertyInfoExtractor<Symfony\\Component\\PropertyInfo\\PropertyInfoExtractor>` class.
+
+.. _`components-property-info-type`:
+
+Type Objects
+------------
+
+Compared to the other extractors, type information extractors provide much more information than can be represented as
+simple scalar values - because of this, type extractors return an array of objects. The array will contain an instance
+of the :class:`Type<Symfony\\Component\\PropertyInfo\\Type>` class for every parameter of a property's mutator method.
+
+.. If you're getting the "type" of a property, should an array of types be returned? If a mutator requires more than
```

Comment:
* An array of types is returned to support properties with several types like `int|float`
* Yes, only mutators with one **required** parameter are taken into account and IIRC it's the type of the required parameter that is used